doc(sc-service-now-incident-events): enhance documentation and add french version CTOR-947 (#4349)
centreon/centreon-documentation: versioned_docs/version-24.10/integrations/event-management/sc-service-now-incident-events.md

---
id: sc-service-now-incident-events

> title: ServiceNow Incident Event

Author: ```suggestion
title: ServiceNow Incident Events
```

---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> The ServiceNow Incident Events stream connector allows you to send data from Centreon to ServiceNow Incident instances.

Reviewer: same as FR

## Before starting

- In most cases, you will want to send data from the central server. It is also possible to send it from a remote server or a poller 
(e.g. if you want to avoid the central server being a SPOF, or if you are an MSP and you install the stream connector on a poller or a remote server within your customer's infrastructure).
- By default, the ServiceNow Incident Events stream connector sends data from [**host_status**](../../developer/developer-broker-mapping.md#host-status) and 
[**service_status**](../../developer/developer-broker-mapping.md#service-status) Broker events. The event format is shown **[here](#event-format)**.

> - These events are sent each time a host or a service is checked. Various parameters let you [filter out events](#filtering-or-adapting-the-data-you-want-to-send-to-servicenow).

Reviewer: ```suggestion
- These events are sent each time a host or a service is checked. Various parameters let you [filter out events](#filtering-or-adapting-the-data-you-want-to-send-to-servicenow-incident).
```

## Installation

Perform the installation on the server that will send data to ServiceNow Incident (central server, remote server, poller).

> 1. Login as `root` using your favorite SSH client.

Reviewer: ```suggestion
1. Log in as `root` using your favorite SSH client.
```

2. Click on **central-broker-master** (or the appropriate broker configuration if it is a poller or a remote server that will send events).
3. On the **Output** tab, select **Generic - Stream connector** from the list and then click **Add**. A new output appears in the list.
4. Fill in the fields as follows:

| Field           | Value                                                               |
| --------------- |---------------------------------------------------------------------|
| Name            | ServiceNow Incident events                                          |
| Path            | /usr/share/centreon-broker/lua/servicenow-incident-events-apiv2.lua |
| Filter category | Neb                                                                 |

5. To enable Centreon to connect to your ServiceNow Incident equipment, fill in the following mandatory parameters. 
The fields for the first entry are already present. Click on the **+Add a new entry** link located below the **Filter category** table to add another one.

| Type   | Name          | Value explanation                    | Value exemple |
| ------ |---------------|--------------------------------------|---------------|

> | string | instance      | the name of the service now instance | MyCompany     |

Reviewer: ```suggestion
| string | instance      | the name of the ServiceNow instance | MyCompany     |
```

| string | client_id     | The Oauth client_id                  | Client_ID     |
| string | client_secret | The Oauth client_secret              | Client_Secret |
| string | username      | The Oauth user                       | User          |

> | string | password      | The Oauth pasword                    | Password      |

Reviewer: ```suggestion
| string | password      | The Oauth password                    | Password      |
```

6. Fill in any optional parameters you want (using the **+Add a new entry** link):

| Type   | Name            | Value explanation                          | Default value                                           |
| ------ |-----------------|--------------------------------------------|---------------------------------------------------------|
| string | logfile         | The file in which logs are written         | /var/log/centreon-broker/servicenow-incident-events.log |
| number | log_level       | Logging level from 1 (errors) to 3 (debug) | 1                                                       |

> | string | http_server_url | The address of the service-now server      | service-now.com                                         |

Reviewer: ```suggestion
| string | http_server_url | The address of the ServiceNow server      | service-now.com                                         |
```

| string | incident_table  | The name of the incident table             | incident                                                |
| string | source          | The source name of the incident            | centreon                                                |

> 7. Use the stream connector's optional parameters to [filter or adapt the data you want Centreon to send to ServiceNow Incident](#filtering-or-adapting-the-data-you-want-to-send-to-servicenow).

Reviewer: ```suggestion
7. Use the stream connector's optional parameters to [filter or adapt the data you want Centreon to send to ServiceNow Incident](#filtering-or-adapting-the-data-you-want-to-send-to-servicenow-incident).
```

## Event format

> This stream connector will send event with the following format.

Reviewer: ```suggestion
This stream connector will send events with the following format.
```
